usage_examples/streamlit/streamlit_app.py

```diff
@@ -0,0 +1,46 @@
+from spacy import util
+from spacy_streamlit import visualize_ner, visualize_textcat
+import streamlit as st
+import os
+
+DEFAULT_TEXT = "Ernest Hemingway, born in Illinois, is generally considered one of the best authors of his time."
+
+st.title("spacy-llm Streamlit Demo")
+st.markdown(
+    """
+    The [spacy-llm](https://github.com/explosion/spacy-llm) package integrates 
+    Large Language Models (LLMs) into spaCy, featuring a modular system 
+    for fast prototyping and prompting, and turning unstructured responses 
+    into robust outputs for various NLP tasks, no training data required.
+
+    This demo uses the OpenAI backend to demonstrate the NER and textcat 
+    tasks.
+    """
+)
+os.environ["OPENAI_API_KEY"] = st.text_input("Your OpenAI API key", type="password")
+text = st.text_area("Text to analyze", DEFAULT_TEXT, height=70)
+
+if os.environ["OPENAI_API_KEY"]:
```

Comment:
I think this might crash if the key isn't set? (I can't remember exactly how streamlit works here.) Maybe `if "OPENAI_API_KEY" in os.environ:` ... `else` helpful-message?

Reply:
I tested it, it won't run anything in the `if` statement until the key is set, so I think it works?

Reply:
I tried it with one of the `spacy-streamlit` demos and it gives a pretty ugly `KeyError` traceback. I think we can check for the key and provide a clearer error message if it's not set?

Reply:
Updated!

Reply:
Ah, I was pretty confused here about how you set this up, sorry. I guess then you always have `os.environ["OPENAI_API_KEY"]`, but it might be `""` or `None` and that's what your `if` is checking for?

Reply:
Yes I think so? I'm not entirely sure though. I think the `else` statement is good anyways.

Reply:
I think the `else` statement is good.

Can you use the current `OPENAI_API_KEY` as the default value for this field, something like `os.environ.get("OPENAI_API_KEY", "")`?

Reply:
Fixed! Anything else?